proof_generation/helper/co_gateway.js

```diff
@@ -0,0 +1,441 @@
+// Copyright 2019 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+const EventDecoder = require('../../test/test_lib/event_decoder');
+
+/**
+ * Object contains input parameter for confirmStakingIntent.
+ *
+ * @typedef {Object} Params
+ * @property {string} params.staker Staker address.
+ * @property {string} params.nonce Staker nonce.
+ * @property {string} params.beneficiary Beneficiary address.
+ * @property {string} params.amount Stake amount.
+ * @property {string} params.gasPrice Gas price.
+ * @property {string} params.gasLimit Gas limit.
+ * @property {string} params.hashLock Hash lock
+ * @property {string} params.blockHeight Block height.
+ * @property {string} params.rlpParentNodes RLP encoded proof data.
+ * @property {string} params.unlockSecret Unlock secret.
+ * @property {string} params.facilitator Facilitator address.
+ * @property {string} params.storageRoot Storage root for proof.
+ */
+
+/**
+ * CoGateway Class provides helper methods for stake and mint and redeem and unstake.
+ */
+class CoGateway {
+  /**
+   * @param {Object} registeredContracts All the deployed contracts
+   */
+  constructor(registeredContracts) {
+    this.utilityToken = registeredContracts.mockUtilityToken;
+    this.coGateway = registeredContracts.coGateway;
+    this.owner = registeredContracts.owner;
+    this.address = this.coGateway.address;
+  }
+
+  /**
+   * Initiates the redeem process.
+   *
+   * @param {Object} params.
```

Comment:
params -> Params
Also, Params is very generic, can it be more specific.

Reply:
Yes 👍 . updated it to ConfirmStakeIntentParams which is more specific.